### ISSUE TITLE
Test error on Intel integrated GPU and Beignet driver

### ISSUE THREAD

**Issue by juliohm** [JuliaGPU/OpenCL.jl#122]
Hi,

I only get a single test error on the latest stable `OpenCL.jl` and Beignet, which is great! Can you please help me understand what is the cause of the error? Should the issue be reported on Beignet instead?

```jl
INFO: ======================================================================
                              Running Behavior Tests
      ======================================================================
Test Summary:           | Pass  Total
  OpenCL Hello World Test |    1      1
OpenCL Low Level Api Test: Error During Test
  Got an exception of type OpenCL.cl.CLError outside of a @test
  CLError(code=-30, CL_INVALID_VALUE)
   in macro expansion; at /home/juliohm/.julia/v0.5/OpenCL/test/test_behaviour.jl:84 [inlined]
   in macro expansion; at ./test.jl:672 [inlined]
   in anonymous at ./<missing>:?
   in include_from_node1(::String) at ./loading.jl:488 (repeats 2 times)
   in process_options(::Base.JLOptions) at ./client.jl:262
   in _start() at ./client.jl:318
Test Summary:             | Error  Total
  OpenCL Low Level Api Test |     1      1
ERROR: LoadError: LoadError: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
 in finish(::Base.Test.DefaultTestSet) at ./test.jl:495
 in macro expansion; at ./test.jl:679 [inlined]
 in anonymous at ./<missing>:?
 in include_from_node1(::String) at ./loading.jl:488 (repeats 2 times)
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
while loading /home/juliohm/.julia/v0.5/OpenCL/test/test_behaviour.jl, in expression starting on line 43
while loading /home/juliohm/.julia/v0.5/OpenCL/test/runtests.jl, in expression starting on line 22
================================================================================[ ERROR: OpenCL ]================================================================================

failed process: Process(`/home/juliohm/Projects/julia/usr/bin/julia -Cnative -J/home/juliohm/Projects/julia/usr/lib/julia/sys.so --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=yes --compilecache=yes /home/juliohm/.julia/v0.5/OpenCL/test/runtests.jl`, ProcessExited(1)) [1]
```

**Comment by vchuravy** [comment]
We are adhering to the OpenCL specs in this case see https://www.khronos.org/registry/cl/sdk/1.0/docs/man/xhtml/clCreateContext.html so it might indeed be an error in the implementation.
Can you manually create a device and context? Since it is very early test failure and it is outside of `@test` no further tests are really run.

```
dev = first(cl.devices())
cl.Context(dev)
```

**Comment by juliohm** [comment]
@vchuravy I can create contexts without problems, and all tests in OpenCL pass except this one.

How can I debug this issue further?

**Comment by juliohm** [comment]
Any idea on what is the cause and how can I fix it?

**Comment by juliohm** [comment]
The tests with the low level API are still failing.

**Comment by vchuravy** [comment]
Which version of beignet are you running?

**Comment by juliohm** [comment]
>Name            : beignet-git
Version         : 1.0.0.r823.gb55060c-1
Description     : A GPGPU System for Intel Ivybridge GPUs
Architecture    : x86_64
URL             : http://cgit.freedesktop.org/beignet/
Licenses        : LGPL2.1
Groups          : None
Provides        : opencl-intel  opencl-headers  opencl-headers12
Depends On      : glu  libsm  libxext  mesa  ncurses  ocl-icd  llvm
Optional Deps   : None
Required By     : None
Optional For    : None
Conflicts With  : beignet  opencl-intel  opencl-headers  opencl-headers12
Replaces        : beignet  opencl-intel  opencl-headers  opencl-headers12
Installed Size  : 36.78 MiB
Packager        : Unknown Packager
Build Date      : Sun 24 Jul 2016 01:29:11 PM BRT
Install Date    : Sun 24 Jul 2016 01:29:32 PM BRT
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : None

**Comment by vchuravy** [comment]
Can you upgrade to version 1.1 or version 1.2?

**Comment by juliohm** [comment]
@vchuravy I have it installed from the AUR `beignet-git` package. It clones the repository and grabs the latest release. I just tried to reinstall it again and got the same version as above. Are 1.1 and 1.2 tagged in the Beignet repository as stable releases?

**Comment by juliohm** [comment]
I am getting a more serious error now with `Beignet 1.0.0.r1062.gab3925f-1`:

>Pkg.test("OpenCL")
INFO: Testing OpenCL
: CommandLine Error: Option 'help-list' registered more than once!
LLVM ERROR: inconsistency in registered CommandLine options
================================================================================[ ERROR: OpenCL ]================================================================================
>
>failed process: Process(`/home/juliohm/Projects/julia/usr/bin/julia -Cnative -J/home/juliohm/Projects/julia/usr/lib/julia/sys.so --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=yes --compilecache=yes /home/juliohm/.julia/v0.5/OpenCL/test/runtests.jl`, ProcessExited(1)) [1]
>
>=================================================================================================================================================================================
INFO: No packages to install, update or remove
ERROR: OpenCL had test errors
 in #test#61(::Bool, ::Function, ::Array{AbstractString,1}) at ./pkg/entry.jl:740
 in (::Base.Pkg.Entry.#kw##test)(::Array{Any,1}, ::Base.Pkg.Entry.#test, ::Array{AbstractString,1}) at ./<missing>:0
 in (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#test,Tuple{Array{AbstractString,1}}})() at ./pkg/dir.jl:31
 in cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#test,Tuple{Array{AbstractString,1}}}, ::String) at ./file.jl:59
 in #cd#1(::Array{Any,1}, ::Function, ::Function, ::Array{AbstractString,1}, ::Vararg{Array{AbstractString,1},N}) at ./pkg/dir.jl:31
 in (::Base.Pkg.Dir.#kw##cd)(::Array{Any,1}, ::Base.Pkg.Dir.#cd, ::Function, ::Array{AbstractString,1}, ::Vararg{Array{AbstractString,1},N}) at ./<missing>:0
 in #test#3(::Bool, ::Function, ::String, ::Vararg{String,N}) at ./pkg/pkg.jl:258
 in test(::String, ::Vararg{String,N}) at ./pkg/pkg.jl:258

**Comment by juliohm** [comment]
Do you know the source of the error already? What can I do to fix it?

**Comment by vchuravy** [comment]
I am travelling until Christmas, so I won't be able to look at it until after then.

Please do upgrade to a stable version and not to a `-git` version.

**Comment by juliohm** [comment]
@vchuravy the stable version does not work well, the git version is not working after this last update. Happy Christmas! Please let me know when you are back.

**Comment by vchuravy** [comment]
This might be fixed by #130

**Comment by yuyichao** [comment]
Correct, this is the test failure I saw.
